### PR TITLE
Update kube provider #478

### LINKED ISSUE #478: Update Kubernetes provider to new config system
Reported by @ryanchard.

We forgot to update the Kubernetes provider with the 0.6 changes.

### PR DESCRIPTION
**Pull request by ZhuozhaoLi**
Fix #478

### CODE REVIEW

**Comment by ZhuozhaoLi** [comment]
> iku

Yes, the main change was tested on Petrelkube before. Today i added the new "overrides" keyword arg and tested on minikube.

File: parsl/providers/kubernetes/kube.py [hunk 1]
```diff
@@ -1,86 +1,107 @@
...
-from parsl.providers.kubernetes.template import template_string
-from parsl.providers.error import OptionalModuleMissing
-
+from parsl.utils import RepresentationMixin
 
 try:
-    from kubernetes import client
-    from kubernetes import config as kube_config
+    from kubernetes import client, config
+    config.load_kube_config()
```
Comment (yadudoc) [inline parsl/providers/kubernetes/kube.py]: Do we need to load the kube config here ? This try catch will set _kube_enabled to false for 2 different reasons and the user won't know if there was just a problem with the load_kube_config.

Comment (ZhuozhaoLi) [reply parsl/providers/kubernetes/kube.py]: Moved `config.load_kube_config()` to init function in KubernetesProvider

**Comment by ZhuozhaoLi** [comment]
Moved `config.load_kube_config()` to init function in KubernetesProvider.
load_kube_config in kubernetes lib will catch the config's exception.

**Pull request merged**